plugins/llm-wiki/hooks/wiki_session_start_drain.sh

```
@@ -0,0 +1,76 @@
+# (POSIX) for the per-cwd marker, JSON encoding via parameter expansion (no jq dep
+# for the emit path).
+
+set -u
+exec 2>/dev/null  # discard stderr — hooks should never spam the user
+
+[[ "${WIKI_SESSION_DRAIN:-1}" == "0" ]] && exit 0
+
+input_json=$(cat 2>/dev/null || true)
+if [[ -n "$input_json" ]] && command -v jq >/dev/null 2>&1; then
+  cwd=$(printf '%s' "$input_json" | jq -r '.cwd // empty' 2>/dev/null)
+  [[ -n "$cwd" && -d "$cwd" ]] && cd "$cwd"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)